### ISSUE TITLE
Compiler crash for Angular @1.6.0-dev.20150729

### ISSUE THREAD

**Issue by alexeagle** [Microsoft/TypeScript#4085]
Thanks for publishing the nightly builds, that is very convenient.
I'm working to update angular (https://github.com/angular/angular/pull/3400)

Repro:
```
$ git clone git@github.com:angular/angular.git; cd angular
$ git fetch origin pull/3400/head:ts1.6
$ git checkout ts1.6
$ npm i
$ cd modules/angular2
$ ../../node_modules/.bin/tsc
/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:29059
                throw e;
                      ^
TypeError: Cannot read property 'kind' of undefined
    at emitSerializedTypeNode (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:27103:29)
    at emitSerializedTypeOfNode (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:27087:25)
    at emitSerializedParameterTypesOfNode (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:27226:37)
    at emitSerializedTypeMetadata (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:27259:25)
    at emitDecoratorsOfConstructor (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:26943:17)
    at emitDecoratorsOfClass (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:26921:17)
    at emitClassLikeDeclarationBelowES6 (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:26879:17)
    at emitClassLikeDeclaration (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:26751:21)
    at emitClassDeclaration (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:26747:24)
    at emitJavaScriptWorker (/Users/alexeagle/Projects/tmp/angular/node_modules/typescript/bin/tsc.js:28675:32)
```

I can also repro with 1.6.0-dev.20150723 FWIW

**Comment by DanielRosenwasser** [comment]
I think this is a duplicate of #3932.

**Comment by DanielRosenwasser** [comment]
@alexeagle for what it's worth, I think you can temporarily get around this by adding a type annotation to your decorated methods

**Comment by mhegazy** [comment]
can you give latest a try?

**Comment by DanielRosenwasser** [comment]
@mhegazy means `master`, we don't have a nightly built yet.

**Comment by alexeagle** [comment]
And didn't update LKG either :)
trying now

**Comment by alexeagle** [comment]
Fixed with current HEAD (and presumably with 20150731 nightly)

**Comment by mhegazy** [comment]
thanks for confirming!
